flow/src/main/java/com/sequenceiq/flow/service/FlowService.java

```diff
@@ -260,4 +268,57 @@ public Optional<OperationFlowsView> getLastFlowOperationByResourceCrn(String res
         List<FlowLog> flowLogs = flowLogDBService.getLatestFlowLogsByCrnInFlowChain(resourceCrn);
         return flowOperationStatisticsService.createOperationResponse(resourceCrn, flowLogs);
     }
+
+    public List<FlowLogResponse> getFlowLogsByIds(List<String> flowIds, Pageable pageable) {
+        LOGGER.info("Getting flow logs by flow ids list {}", flowIds);
+        return flowLogDBService.getFlowLogsByFlowIdsCreatedDesc(new HashSet<>(flowIds), pageable)
+                .stream().map(flowLog -> flowLogConverter.convert(flowLog)).collect(toList());
+    }
+
+    public List<FlowCheckResponse> getFlowChainsByChainIds(List<String> chainIds, Pageable pageable) {
+        LOGGER.info("Getting flow check responses by flow check ids list {}", chainIds);
+        Map<String, FlowChainLog> flowChainsMap = flowChainLogService
+                .findAllByFlowChainIdInOrderByCreatedDesc(new HashSet<>(chainIds), pageable)
+                .stream().collect(Collectors.toMap(FlowChainLog::getFlowChainId, Function.identity()));
+
+        // GETTING RELATED FLOW CHAIN LOGS
+        Map<String, List<FlowChainLog>> relatedFlowChainsMap = new HashMap<>();
+        Map<String, String> parentMap = new HashMap<>();
+        flowChainsMap.forEach((key, flowChainLog) -> {
+            List<FlowChainLog> relatedFlowChainLogs = flowChainLogService.getRelatedFlowChainLogs(List.of(flowChainLog));
```

Comment:
why can't we fetch with all the chainlog ids at once? if we have 2 chainds which have the same parent chain, it would fetch everything twice, even if they are the same

Reply:
I need to associate which flow log is related to which parent chain id so that i can calculate the correct end time and if the flow chain has any in pending state. For that I have to create a parent child relationship and that is why I have to send in the flow chains one by one to get related flow chain logs.